### ISSUE TITLE
Hierarchical editing of models: missing 'each' when modifying array components

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#13126]
Follow-up of [#2891 (comment)](https://github.com/OpenModelica/OpenModelica/issues/2891#issuecomment-2441996977) 

I played around a bit with this model:
```modelica
model TestEditing
  extends SOFCPoliMi.Tests.BenchmarkCammarata;
annotation(uses(SOFCPoliMi));
end TestEditing;
```
which is definitely non-trivial. In general it works quite well, I like it.

I found an issue with modifiers applied to sub-components of arrays, that I replicated with this simple MWE:

```modelica
package TestArrayModifiers
    model System
  Module modules[2] annotation(
      Placement(transformation(origin = {-18, 0}, extent = {{-10, -10}, {10, 10}})));
  equation

  end System;

  model Module
  Component component annotation(
      Placement(transformation(origin = {-10, 4}, extent = {{-10, -10}, {10, 10}})));
  equation

  end Module;

  model Component
  parameter Real p = 1;
    SubComponent subComponents[2] annotation(
      Placement(transformation(origin = {2, -6}, extent = {{-10, -10}, {10, 10}})));
  end Component;

  model SubComponent
    parameter Real q = 1;
  end SubComponent;

  model SystemVariant1
    extends System;
  end SystemVariant1;

  model SystemVariant2
    extends System;
  end SystemVariant2;

end TestArrayModifiers; 
```
Steps to reproduce the issue:
- open `SystemVariant1`
- do Show Elements on `modules`
- double-click on `component`
- change the value of parameter `p` to 2

The following code is created:
```modelica
model SystemVariant1
  extends System(modules(component(p = 2)));
end SystemVariant1;
```
which is invalid:
```
[2] 16:41:07 Translation Notification
[TestArrayModifiers: 21:38-21:43]: From here:

[3] 16:41:07 Translation Error
[TestArrayModifiers: 17:5-17:25]: Non-array modification '2' for array component 'p', possibly due to missing 'each'.
```
The correct code should be:
```modelica
model SystemVariant1
  extends System(modules(each component(p = 2)));
end SystemVariant1;
```
Now do the following steps:
- open `SystemVariant2`
- do Show Elements on `modules`
- do Show Elements on `component`
- double-click on `subComponents`
- change the value of parameter `q` to 3

The following code is created:
```modelica
model SystemVariant2
  extends System(modules(component(subComponents(q = 3))));
end SystemVariant2;
```
which is also invalid:
```
[1] 16:55:58 Translation Notification
[TestArrayModifiers: 31:52-31:57]: From here:

[2] 16:55:58 Translation Error
[TestArrayModifiers: 23:5-23:25]: Non-array modification '3' for array component 'q', possibly due to missing 'each'.
```
The correct code should be 
```modelica
model SystemVariant2
  extends System(modules(each component(subComponents(each q = 3))));
end SystemVariant2;
```

In other words, whenever you do Show Element on an array (`modules`, and `subComponents` in this case), all its corresponding modifiers should get an `each` prefix to be valid.

**Comment by casella** [comment]
Great, thanks!
